controllers/ibu_controller.go

```diff
@@ -121,16 +123,105 @@ func (r *ImageBasedUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	} else if reconcileTime == utils.StopReconciling {
 		return
 	}
+
+	if imageBasedUpgrade.Spec.Stage == ranv1alpha1.Stages.Idle {
+		return
+	}
+
+	ibus := &ranv1alpha1.ImageBasedUpgradeList{}
+	err = r.Client.List(context.TODO(), ibus)
+	if err != nil {
+		return
+	}
+
+	for _, ibu := range ibus.Items {
+		if ibu.Name != imageBasedUpgrade.Name && isIbuActive(&ibu) {
+			r.setBlockingCondition(imageBasedUpgrade, &ibu)
+			nextReconcile = requeueWithLongInterval()
+			err = r.updateStatus(ctx, imageBasedUpgrade)
+			return
+		}
+	}
+
+	previousSucceededCondition := utils.GetPreviousSucceededCondition(imageBasedUpgrade)
+	previousPendingCondition := utils.GetPreviousPendingCondition(imageBasedUpgrade)
+
+	if imageBasedUpgrade.Spec.Stage == ranv1alpha1.Stages.Rollback {
```

Comment:
I guess this block is doing a "state transition validation" check? Probably could be in its own function

Reply:
Yes, I was planning to do that and some unit testing for the function.

Reply:
Updated